gpjax/linops/linear_operator.py

```diff
@@ -35,8 +30,9 @@
 NestedT = Union[T, Iterable["NestedT"], Mapping[Any, "NestedT"]]
 
 # Nested types.
+DTypes = Union[Type[jnp.float32], Type[jnp.float64], Type[jnp.int32], Type[jnp.int64]]
 ShapeT = TypeVar("ShapeT", bound=NestedT[Tuple[int, ...]])
-DTypeT = TypeVar("DTypeT", bound=NestedT[jnp.dtype])
+DTypeT = TypeVar("DTypeT", bound=NestedT[DTypes])
```

Comment:
Could these be moved into `linops.utils` to be consistent with other types?

Reply:
it's not linops-specific so I would actually favour moving it into something like gpjax.types

Reply:
Sure, let's do that.